testthat 3e

dpprdan opened the pull request:
## Description

This PR makes the necessary changes to use [testthat 3e](https://testthat.r-lib.org/articles/third-edition.html). Builds on #130. 

- [ ] Wait for vcr v0.7
- [ ] Add NEWS item

testthat 3e [does not suppress messages anymore](https://testthat.r-lib.org/articles/third-edition.html), so one gets [a lot of messages from vcr](https://github.com/ropensci/opencage/pull/135/checks#step:11:33) like the following

```
CrulAdapter enabled!
HttrAdapter enabled!
net connect allowed
ejecting cassette: NAME_OF_CASSETTE
CrulAdapter disabled!
HttrAdapter disabled!
```

~~I haven't found a vcr setting to supress these when running tests, yet (except `quiet` which seems to be a [per-cassette setting](https://docs.ropensci.org/vcr/reference/vcr_configure.html#arguments)).~~ 

vcr messages can be supressed with the new `vcr_configure(quiet = TRUE)` as of vcr v0.7.

dpprdan committed:
start with vcr

dpprdan committed:
vcr oc_get

dpprdan committed:
vcrify all the things

dpprdan committed:
mind your lint

dpprdan committed:
skip_if_no_key and match on uri only

dpprdan committed:
memoise::forget

dpprdan committed:
vcr-mask OpenCage test keys as well so we can identify them in cassettes

dpprdan committed:
lint

dpprdan committed:
add vcr's .gitattributes

dpprdan committed:
rename helper- to setup-opencage.R, use vcr_test_path() and check_cassette_names()

dpprdan committed:
use vcr with tests of deprecated functions, delete duplicate tests in test-opencage_forward.R

dpprdan committed:
re-record vcr_cassettes

dpprdan committed:
check if a key is present before sending a query

dpprdan committed:
rename helper-opencage.R to setup-opencage.R

dpprdan committed:
add NEWS item

dpprdan committed:
set fake API key and configure rate limit for vcr

dpprdan committed:
test against default rate limit

dpprdan committed:
drop skip_if_no_key() and skip_of_oc_offline()

dpprdan committed:
vcrify oc_get() tests

dpprdan committed:
Merge branch 'master' into feat_vcr

dpprdan committed:
reduce rate limit with vcr also when key is available

dpprdan committed:
split up large cassettes and other style improvements

dpprdan committed:
mention vcr in contributing.md

dpprdan committed:
moar cassettes

dpprdan committed:
bump required memoise version because https://github.com/r-lib/memoise/pull/61

dpprdan committed:
Merge branch 'master' into feat_vcr

dpprdan committed:
use testthat3e

dpprdan committed:
expect_is() is deprecated

dpprdan committed:
test lifecycle::expect_deprecated only once

because testthat3e does not ignore messages anymore https://testthat.r-lib.org/articles/third-edition.html#messages

maelle commented:
reg vcr messages cf https://github.com/ropensci/vcr/issues/226 by @sckott thanks to a report by @stitam

dpprdan commented:
💯 I figured that this was already solved somehow and that my google fu just failed me somehow.

dpprdan committed:
bump suggested vcr version to current dev

dpprdan committed:
re-use cassettes and delete empty ones

dpprdan committed:
clear cache before running tests

dpprdan committed:
Merge branch 'main' into feat_vcr

dpprdan committed:
re-record cassette for min_confidence test

dpprdan committed:
configure OS and RSPM only once in matrix, include OS version in cache key

dpprdan committed:
OPENCAGE_KEY not necessary in R-CMD-check workflow anymore

all API requests run against vcr cassettes; a fake key is provided in setup_opencage.R

dpprdan committed:
add weekly test with real API requests

dpprdan committed:
add NEWS item

dpprdan committed:
Merge pull request #137: test with vcr

dpprdan committed:
bump required testthat version to 3.0.0

dpprdan commented:
closed in favour of #138
